data/src/isupport.rs

```diff
@@ -139,12 +140,11 @@ impl FromStr for Operation {
                             parse_required_positive_integer(value)?,
                         ))),
                         "CHANTYPES" => {
-                            if value.is_empty() {
+                            let chars = value.chars().collect::<Vec<_>>();
+                            if chars.is_empty() {
                                 Ok(Operation::Add(Parameter::CHANTYPES(None)))
-                            } else if value.chars().all(|c| proto::CHANNEL_PREFIXES.contains(&c)) {
-                                Ok(Operation::Add(Parameter::CHANTYPES(Some(
-                                    value.to_string(),
-                                ))))
+                            } else if chars.iter().all(|c| proto::CHANNEL_PREFIXES.contains(c)) {
+                                Ok(Operation::Add(Parameter::CHANTYPES(Some(chars))))
```

Comment:
Why do we need to validate what the server specifies as a channel prefix? In theory a server can choose something completely random and we should respect it? I think we can remove `proto::CHANNEL_PREFIXES` completely since we should just trust the server and if they don't specify ISUPPORT, we use the `DEFAULT_CHANNEL_PREFIXES`.

Same with `CHANNEL_MEMBERSHIP_PREFIXES` below.

Reply:
If we want to open the floodgates for any possible channel/ `STATUSMSG`, then it would probably be good for us to handle the `PREFIX` ISUPPORT parameter.  That gives the mapping between prefix characters and user access levels, and it's my reading that `STATUSMSG` prefixes should all come from the default/provided set of `PREFIX` characters.  Aside from `STATUSMSG`, I believe user access level parsing should be derived from the mapping provided by the `PREFIX` ISUPPORT parameter.